backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -481,15 +483,89 @@ private CreatedAndDuplicatedAndSkippedBranches<NonRootManagedBranchSnapshot> cre
       val customAnnotation = entry.getCustomAnnotation();
       val childBranches = deriveChildBranches(coreLocalBranch, entry.getChildren());
       val remoteTrackingBranch = getRemoteTrackingBranchForCoreLocalBranch(coreLocalBranch);
+
+      val remoteForkPoint = deriveRemoteAwareForkPoint(coreLocalBranch);
+
       val statusHookOutput = statusHookExecutor.deriveHookOutputFor(branchName, pointedCommit);
 
       val result = new NonRootManagedBranchSnapshot(branchName, branchFullName, childBranches.getCreatedBranches(),
-          pointedCommit, remoteTrackingBranch, relationToRemote, customAnnotation, statusHookOutput, forkPoint,
+          pointedCommit, remoteTrackingBranch, relationToRemote, customAnnotation, statusHookOutput, forkPoint, remoteForkPoint,
           commits.map(CommitOfManagedBranch::new), syncToParentStatus);
       return CreatedAndDuplicatedAndSkippedBranches.of(List.of(result),
           childBranches.getDuplicatedBranchNames(), childBranches.getSkippedBranchNames());
     }
 
+    @UIThreadUnsafe
+    private @Nullable ForkPointCommitOfManagedBranch deriveRemoteAwareForkPoint(IGitCoreLocalBranchSnapshot coreLocalBranch)
```

Comment:
Okay... so in a nutshell, how does this algorithm compare to the existing fork point algorithm? 🤔

Reply:
it only checks the reflog of the remote

Reply:
the current implementation is a tad bit inefficient though.
There is a bug in the refactoring of the existing refactor functionality as well. I would look into that tomorrow.